### ISSUE TITLE
Audio / frame rates slowly degrade over time and only fixed by a system reboot

### ISSUE THREAD

**Issue by cmcginty** [libretro/RetroArch#5839]
Not sure if this is a core issue or what. Just creating a ticket in case others are have the same problems.

## Description

This is a weird issue where I noticed in retroarch that after loading/unloading multiple cores or the same cores the audio quality starts getting more glitches and framerates slow down even more.

This issue seems to be at the OS level or below because restarting retroarch does not resolve the degraded performance the only workaround is to do a complete restart of my device (Asus Chromebox running Lubuntu 17.04)

I usually hear audio pops when starting a core for the first time. It might be related to the issue, or not.

### Steps to reproduce the bug

The bug is most reproducible with ParaLLEl N64 but only because I think this is running my hardware at the edge.

1. Continuously load/close different cores or same care (i.e. ParaLLEl N64)
2. After 20 to 30 core loads the audio continues to get worse and the frame rates drop from near 57-58fps to 53fps

I haven't try to reload the audio modules or make any other OS config changes, but I might try that next to see if I can narrow down which sub-system is causing the issue.

### Bisect Results

I've noticed this all year but only now realized that loading the cores seem to be causing the issue.

### Environment information

- OS: Lubuntu 17.04
- Hardware: Asus Chromebox, HDMI audio and video
- Game: Mario 64

### Keywords
audio degrade, stutter, glitch, reload, reboot, restart resolves

**Comment by Oibaf66** [comment]
Check the satus of the used memory (RAM). It could depend on a leak memory issue. 

https://en.wikipedia.org/wiki/Memory_leak

**Comment by meepingsnesroms** [comment]
Its not a memory leak, whether you have 3gb or 1gb of ram available as long as there is the same amount the console has or more it wont matter.

> restarting retroarch does not resolve the degraded performance

Then it cant be fixed by RetroArch, I will still help you find the cause and fix it though.

What other programs are running?
Can you get a full list before and after the issue, including memory used by each program, cpus used and at what percentage by each program along with total threads running.

**Comment by cmcginty** [comment]
It's a dedicated box for my TV. There is nothing else running the foreground except the X server window (Openbox WM) and login managers (lightdm).

I can try restarting window manager and X server to see if that fixes it. My next through was try to reload some kernel modules, specifically the audio/video. I'll try and get the list of modules.

**Comment by meepingsnesroms** [comment]
Is RetroArch running through xwindows?

All I can think of is loading/unloading a core reinitializes the audio/video drivers so the same result should be achieved by loading/unloading RetroArch or any other fullscreen program 30 times.

**Comment by cmcginty** [comment]
Right it's running from the desktop. Is there another method?

**Comment by meepingsnesroms** [comment]
Is it fullscreen when you launch it, what is your video driver under Settings->Drivers?
You can push left and right to pick a new graphics driver if its supported just dont choose "null" or it wont load anymore.

**Comment by fr500** [comment]
@meepingsnesroms of course it matters, because the mupen core particularly leaks **a lot** when performing savestates.

If rewind is enabled it can lead to a BSOD even in a very high end rig.

**Comment by fr500** [comment]
The serialization code leaks that's what I can tell you

**Comment by meepingsnesroms** [comment]
Do any other cores have this issue or just mupen?

**Comment by rz5** [comment]
Yes, would be important to know if this happens to other cores. It may be due to the undo save/load stuff that I wrote in RA.

**Comment by fr500** [comment]
No it's not due to undo save state it's due to the serializer in the core.

**Comment by LibretroAdmin** [comment]
Lacks too much information to do anything with. But I can probably investigate later if there is anything to this
